silk/datadog_checks/silk/check.py

```diff
@@ -0,0 +1,224 @@
+        else:
+            self.log.debug("Could not submit version metadata, got: %", version)
+
+    def _assign_host_tags(self, state_data):
+        self._system_tags = []
+        self._system_tags.append('system_name:{}'.format(state_data.get('system_name')))
+        self._system_tags.append('system_id:{}'.format(state_data.get('system_id')))
+
+    def parse_metrics(self, output, path, tags=None, metrics_mapping=None, get_method=None):
+        """
+        Parse metrics from HTTP response. return_first will return the first item in `hits` key.
+        """
+        if not output:
+            self.log.debug("No results for path %s", path)
+            return
+
+        if not tags:
+            tags = self._tags
+
+        for item in output:
+            metric_tags = deepcopy(tags)
+
+            for key, tag_name in metrics_mapping.tags.items():
+                if key in item:
+                    metric_tags.append("{}:{}".format(tag_name, item.get(key)))
+
+            for key, metric in metrics_mapping.metrics.items():
+                metric_part = None
+                raw_metric_name, method = metric
+
+                if metrics_mapping.field_to_name:
+                    for field, name_map in metrics_mapping.field_to_name.items():
+                        metric_part = name_map.get(item.get(field))
+                if key in item:
+                    if metric_part:
+                        name = "{}.{}.{}".format(metrics_mapping.prefix, metric_part, raw_metric_name)
+                    else:
+                        name = "{}.{}".format(metrics_mapping.prefix, raw_metric_name)
+
+                    get_method(self, method)(name, item.get(key), tags=metric_tags)
+
+    def _get_data(self, path):
+        url = urljoin(self.url, path)
+        try:
+            self.log.debug("Trying to get metrics from %s", url)
+            response = self.http.get(url)
+            response.raise_for_status()
+            response_json = response.json()
+            code = response.status_code
+            if response_json:
+                if 'error_msg' in response_json:
+                    msg = "Received error message: %s", response_json.get('error_msg')
+                    self.log.warning(msg)
+                    self.service_check(self.CONNECT_SERVICE_CHECK, AgentCheck.WARNING, message=msg, tags=self._tags)
+                    return None, code
+            return response_json.get("hits"), code
+        except Exception as e:
+            self.log.debug("Encountered error while getting metrics from %s: %s", path, str(e))
+            raise
+
+    def collect_events(self, tags):
+        self.log.debug("Starting events collection (query start time: %s).", self.latest_event_query)
+        latest_event_time = None
+        collect_events_start_time = get_timestamp()
+        try:
+            event_query = EVENT_PATH.format(self.latest_event_query)
+            raw_events, code = self._get_data(event_query)
```

Comment:
What events are returned when calling this API with a specific timestamp?

Reply:
The `EVENT_PATH` value is `events?timestamp__gte={}`, and from the [Silk API docs for events](https://github.com/silk-us/silk-sdp-api-docs/blob/master/events.md) `gte` is used to get only values greater than the value inputted (in this case `latest_event_query`). So this API call should return all events with timestamps greater than `self.latest_event_query`. `self.latest_event_query` is updated after `collect_events()` to avoid collecting the same events in subsequent runs.

Reply:
Gotcha, thanks for the explanation